packages/manager/modules/pci/src/projects/project/instances/instances.routing.js

```diff
@@ -170,7 +187,82 @@ export default /* @ngInject */ ($stateProvider) => {
         OvhApiCloudProjectInstance,
       ) => () => {
         OvhApiCloudProjectInstance.v6().resetAllCache();
-        $state.reload();
+        return $state.reload();
+      },
+      killTasks: /* @ngInject */ (Poller) => (pattern) => Poller.kill(pattern),
+
+      /** If one/some instances is still in shelving we run a sub task until the instance is shelved
+          Also, the sub tasks are killed if changed state is different from instances.*
+      */
+      pollShelvingInstances: /* @ngInject */ (
+        $translate,
+        CucCloudMessage,
+        Poller,
+        instances,
+        projectId,
+        refreshInstances,
+      ) => {
+        instances.forEach((instance) => {
```

Comment:
Can `instances` be mapped toan Array of `Instance` (class), before to call instance methods ?

Reply:
Instances is already an array of `Instance`.

Reply:
Hmm so why do you need to reinstanciate them ? (L206)

Reply:
May be you discussed on an older snippet code version.
But actually i don't instantiate them.

Reply:
@mohammed-zahaf I think what @JeremyDec meant is line 211 https://github.com/ovh/manager/pull/4401/files#diff-95a5ce8093382673d816afec8e5c4c5e02631db7b83fa14409efc0f4940adba6R211